kfs-core/src/main/java/edu/arizona/kfs/gl/batch/service/impl/BudgetAdjustmentCashTransferServiceImpl.java

```diff
@@ -0,0 +1,596 @@
+    }
+        
+    protected void createOutputEntry(OriginEntryInformation oef, PrintStream group) throws IOException {       
+        try {
+            group.printf("%s\n", oef.getLine());
+        }
+        catch (Exception e) {
+            throw new IOException(e.toString());
+        }
+    }
+    
+    protected void createOutputEntry(String line, PrintStream ps) throws IOException {
+        try {
+            ps.printf("%s\n", line);
+        } catch (Exception e) {
+            throw new IOException(e.toString());
+        }
+    }
+    
+    /**
+     * This will determine if this transaction is a budget adjustment transaction
+     * 
+     * @param transaction the transaction which is being determined to be a budget adjustment transaction
+     * @return true if the transaction is a budget adjustment transaction and therefore should have a cash transfer transaction created for it; false if otherwise
+     */
+    public boolean isBudgetAdjustmentTransaction(OriginEntryFull transaction) {
+    	    	
+    	//retrieve document type and origination code parameter values
+    	Collection<String> documentTypeCodes = parameterService.getParameterValuesAsString(PosterEntriesStep.class, GeneralLedgerConstants.CASH_TRANSFER_DOC_TYPE_CODES);
+    	Collection<String> originationCodes = parameterService.getParameterValuesAsString(PosterEntriesStep.class, GeneralLedgerConstants.CASH_TRANSFER_ORIGINATION_CODES);
+      
+        //get current year system options
+    	SystemOptions options = getOptionsService().getOptions(universityDateService.getCurrentFiscalYear());
+        if (ObjectUtils.isNull(options)) {
+        	options = getOptionsService().getCurrentYearOptions();
+        }
+         
+        // Does the balance type code indicate that this transaction is for current budget?
+        if (!transaction.getFinancialBalanceTypeCode().equals(options.getBudgetCheckingBalanceTypeCd())) {
+        	return false;
+        }                   
+               
+        // Does the origin code indicate that this transaction is from the Budget Office, or ERE Sweep?
+        if (!originationCodes.contains(transaction.getFinancialSystemOriginationCode())) {
+        	 return false;
+        }                   
+        
+        // Is the document type correct?
+        if (!documentTypeCodes.contains(transaction.getFinancialDocumentTypeCode())) {
+        	return false;
+        }           
+
+        return true;  // still here?  then I guess we don't have an exclusion
+       
+    }
+    
+    protected String getFinancialObjectTypeCode(Integer universityFiscalYear, String chartOfAccountsCode, String financialObjectCode) {    
+    	String financialObjectTypeCode = null;
+    	
+	    ObjectCode objectCode = getObjectCodeService().getByPrimaryId(universityFiscalYear, chartOfAccountsCode, financialObjectCode);
+	    if (ObjectUtils.isNotNull(objectCode)) {   	    
+	    	financialObjectTypeCode = objectCode.getFinancialObjectTypeCode();
```

Comment:
If this method is called multiple times with the same params, could be useful caching some of the financialObjectTypeCode.